var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,223 @@
+                    '${LIBXML2_HOME}/lib $ENV{LIBXML2_HOME}/lib',
+                    'CMake/FindLibxml2QMC.cmake')
+
+    def cmake_args(self):
+        args = []
+
+        if '+mpi' in self.spec:
+            mpi = self.spec['mpi']
+            args.append('-DCMAKE_C_COMPILER={0}'.format(mpi.mpicc))
+            args.append('-DCMAKE_CXX_COMPILER={0}'.format(mpi.mpicxx))
+            args.append('-DMPI_BASE_DIR:PATH={0}'.format(mpi.prefix))
+
+        # Currently FFTW_HOME and LIBXML2_HOME are used by CMake.
+        # Any CMake warnings about other variables are benign.
+        xml2_prefix = self.spec['libxml2'].prefix
+        args.append('-DLIBXML2_HOME={0}'.format(xml2_prefix))
+        args.append('-DLibxml2_INCLUDE_DIRS={0}'.format(xml2_prefix.include))
+        args.append('-DLibxml2_LIBRARY_DIRS={0}'.format(xml2_prefix.lib))
+
+        fftw_prefix = self.spec['fftw'].prefix
+        args.append('-DFFTW_HOME={0}'.format(fftw_prefix))
+        args.append('-DFFTW_INCLUDE_DIRS={0}'.format(fftw_prefix.include))
+        args.append('-DFFTW_LIBRARY_DIRS={0}'.format(fftw_prefix.lib))
+
+        args.append('-DBOOST_ROOT={0}'.format(self.spec['boost'].prefix))
+        args.append('-DHDF5_ROOT={0}'.format(self.spec['hdf5'].prefix))
+
+        # Default is MPI, serial version is convenient for cases, e.g. laptops
+        if '+mpi' in self.spec:
+            args.append('-DQMC_MPI=1')
+        elif '~mpi' in self.spec:
+            args.append('-DQMC_MPI=0')
+
+        # Default is real-valued single particle orbitals
+        if '+complex' in self.spec:
+            args.append('-DQMC_COMPLEX=1')
+        elif '~complex' in self.spec:
+            args.append('-DQMC_COMPLEX=0')
+
+        # When '-DQMC_CUDA=1', CMake automatically sets:
+        # '-DQMC_MIXED_PRECISION=1'
+        #
+        # There is a double-precision CUDA path, but it is not as well
+        # tested. 
+
+        if '+cuda' in self.spec:
+            args.append('-DQMC_CUDA=1')
+        elif '~cuda' in self.spec:
+            args.append('-DQMC_CUDA=0')
+
+        # Mixed-precision versues double-precision CPU and GPU code        
+        if '+mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=1')
+        elif '~mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=0')
+
+        # New Structure-of-Array (SOA) code, much faster than default
+        # Array-of-Structure (AOS) code. 
+        # No support for local atomic orbital basis.
+        if '+soa' in self.spec:
+            args.append('-DENABLE_SOA=1')
+        elif '~soa' in self.spec:
+            args.append('-DENABLE_SOA=0')
+
+        # Manual Timers
+        if '+timers' in self.spec:
+            args.append('-DENABLE_TIMERS=1')
+        elif '~timers' in self.spec:
+            args.append('-DENABLE_TIMERS=0')
+        
+        # Include MKL flags     
+        if 'intel-mkl' in self.spec:
+            args.append('-DBLA_VENDOR=Intel10_64lp_seq')
```

Comment:
looks like here you ingnore altogether any options used in `intel-mkl`:
```
    ilp64 [off]       True, False       64 bit integers
    threads [none]    openmp, none      Multithreading support
```
that's why packages should really use `libs()` interface.

Reply:
if you use CMake, can you not explicitly set Blas/Lapack libraries, like for example [here](https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/dealii/package.py#L185-L188). Of course, it depends on how CMake is being used inside...

Reply:
the same is with other Blas/Lapack, say the build system will be free to pick up single threaded or multithreaded Atlas libs, regardless of package settings:
```
threads [none]    pthreads, none    Multithreading support
```

Reply:
I think QMCPACK is not compatible with threaded BLAS. This may be for legacy reason related to the OpenMP runtime which may no longer be true. We only support sequential without 64-bit integers.

Reply:
> We only support sequential without 64-bit integers.

you need to add _conflicts_, something like:

```
conflicts('^intel-mkl threads=pthreads')
```
to make sure users don't use `qmcpack` in a stack of software with multithreaded blas.